packages/machine-extractor/src/__tests__/actions.test.ts

```diff
@@ -0,0 +1,1178 @@
+			  sendTo('actor', 'event', {id: 'namedId'})
+			],
+		  }
+		}
+	  });
+	`);
+
+    expect(config.states!.a.entry).toMatchInlineSnapshot(`
+      {
+        "action": {
+          "event": {
+            "type": "event",
+          },
+          "id": "namedId",
+          "to": "actor",
+          "type": "xstate.sendTo",
+        },
+        "kind": "builtin",
+      }
+    `);
+  });
+
+  it('should extract sendTo with a number id', () => {
+    const config = getTestMachineConfig(`
+	  createMachine({
+		initial: "a",
+		states: {
+		  a: {
+			entry: [
+			  sendTo('actor', 'event', {id: 2.34e2})
+			],
+		  }
+		}
+	  });
+	`);
+
+    expect(config.states!.a.entry).toMatchInlineSnapshot(`
+      {
+        "action": {
+          "event": {
+            "type": "event",
+          },
+          "id": 234,
+          "to": "actor",
+          "type": "xstate.sendTo",
+        },
+        "kind": "builtin",
+      }
+    `);
+  });
+
+  it('should extract sendTo with a callback id', () => {
+    const config = getTestMachineConfig(`
+	  createMachine({
+		initial: "a",
+		states: {
+		  a: {
+			entry: [
+			  sendTo('actor', 'event', {id: () => {}})
+			],
+			exit: [
+			  sendTo('actor', 'event', {id: function() {}})
+			],
+		  }
+		}
+	  });
+	`);
+
+    expect(config.states!.a.entry).toMatchInlineSnapshot(`
+      {
+        "action": {
+          "event": {
+            "type": "event",
+          },
+          "id": "{{() => {}}}",
+          "to": "actor",
+          "type": "xstate.sendTo",
+        },
+        "kind": "builtin",
+      }
+    `);
+    expect(config.states!.a.exit).toMatchInlineSnapshot(`
+      {
+        "action": {
+          "event": {
+            "type": "event",
+          },
+          "id": "{{function() {}}}",
+          "to": "actor",
+          "type": "xstate.sendTo",
+        },
+        "kind": "builtin",
+      }
+    `);
+  });
+
+  it('should extract sendTo with any expressions for id', () => {
+    const config = getTestMachineConfig(`
+	  createMachine({
+		initial: "a",
+		states: {
+		  a: {
+			entry: [
+			  sendTo('actor', 'event', {id: somevar})
+			],
+		  }
+		}
+	  });
+	`);
+
+    expect(config.states!.a.entry).toMatchInlineSnapshot(`
+      {
+        "action": {
+          "event": {
+            "type": "event",
+          },
+          "id": "{{somevar}}",
+          "to": "actor",
+          "type": "xstate.sendTo",
+        },
+        "kind": "builtin",
+      }
+    `);
+  });
+
+  it('Should extract stop with a string id', () => {
+    const config = getTestMachineConfig(`
+	  createMachine({
+		initial: "a",
+		states: {
+		  a: {
+			entry: [
+			  stop('actor')
+			],
+		  },
+		},
+	  });
+	`);
+
+    expect(config.states!.a.entry).toMatchInlineSnapshot(`
+      {
+        "action": {
+          "id": "actor",
+          "type": "xstate.stop",
+        },
+        "kind": "builtin",
+      }
+    `);
+  });
+
+  it('Should extract stop with a number id', () => {
```

Comment:
```suggestion
  it('Should extract stop with a bigint id', () => {
```

Reply:
although, if it's "big enough" it probably wouldn't work at runtime in XState but that's a separate issue~